### PR TITLE
Add disable buffer flag

### PR DESCRIPTION
**Pull request by rzlim08**
It definitely looks as though there is a memory leak being caused by this buffer. This will allow the user to set a flag to disable the buffer.

### CODE REVIEW

File: cmd/consensusGenome/consensusGenome.go [hunk 1]
```diff
@@ -129,6 +130,7 @@ func loadSharedFlags(c *cobra.Command) {
 		nanoporeDefaultWetlabProtocol,
 		defaultMedakaModel,
 	))
+        c.Flags().BoolVar(&disableBuffer, "disable-buffer", false, "Disable shared buffer pool (useful if running out of memory)")
```
Comment (tfrcarvalho) [inline cmd/consensusGenome/consensusGenome.go:133]: nit: fix indentation

File: cmd/consensusGenome/uploadSamples.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ var uploadSamplesCmd = &cobra.Command{
 			WetlabProtocols[wetlabProtocol],
 			MedakaModels[medakaModel],
 			clearLabs,
+                        disableBuffer,
```
Comment (tfrcarvalho) [inline cmd/consensusGenome/uploadSamples.go:46]: nit: fix indentation

File: pkg/idseq/uploadSamplesFlow.go [hunk 3]
```diff
@@ -22,6 +22,7 @@ func UploadSamplesFlow(
 	wetlabProtocol string,
 	medakaModel string,
 	clearLabs bool,
+        disableBuffer bool,
```
Comment (tfrcarvalho) [inline pkg/idseq/uploadSamplesFlow.go:25]: nit: fix indentation

File: pkg/upload/upload.go [hunk 4]
```diff
@@ -59,11 +59,19 @@ func NewUploader(creds aws.Credentials) Uploader {
...
+        if disableBuffer { 
+       		uploader = manager.NewUploader(client, func(u *manager.Uploader) {
+                        u.LeavePartsOnError = true
+                        u.Concurrency = runtime.NumCPU()
+                }) 
+        } else { 
+		uploader = manager.NewUploader(client, func(u *manager.Uploader) {
+			u.LeavePartsOnError = true
+			u.Concurrency = runtime.NumCPU()
+			u.BufferProvider = manager.NewBufferedReadSeekerWriteToPool(int(DefaultUploadPartSize) * runtime.NumCPU())
```
Comment (tfrcarvalho) [inline pkg/upload/upload.go:72]: Can't this be the only conditional line?

**Review by tfrcarvalho — COMMENTED** [review]
Apart from some indentation issues: tabs vs spaces?
It looks good to me.

We might need a check from @tmorse when he comes back, but seems safe enough.

File: pkg/upload/upload.go (hunk 4 above)
Comment (rzlim08) [reply pkg/upload/upload.go:72]: probably, but I'm not sure how exactly go treats "if" statements in lambdas. I'll keep it separate for now.

**Pull request merged**